src/app/(bluerpint-list)/BlueprintList.tsx

```diff
@@ -0,0 +1,126 @@
+'use client';
+
+import Link from 'next/link';
+import BlueprintCard from '@/app/components/BlueprintCard';
+import zkeSdk, { Blueprint } from '@dimidumo/zk-email-sdk-ts';
+import { useState, useEffect, useRef, useCallback } from 'react';
+
+interface BlueprintListProps {
+  search: string | null;
+}
+
+export default function BlueprintList({ search }: BlueprintListProps) {
+  const [blueprints, setBlueprints] = useState<Blueprint[]>([]);
+  const [skip, setSkip] = useState(0);
+  const [limit] = useState(10);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasMore, setHasMore] = useState(true);
+  const [error, setError] = useState<Error | null>(null);
+  const observerRef = useRef<IntersectionObserver>();
+  const loadingRef = useRef<HTMLDivElement>(null);
+
+  const fetchBlueprints = useCallback(async () => {
+    if (isLoading || !hasMore) return;
+
+    setIsLoading(true);
+    setError(null);
+    const sdk = zkeSdk();
```

Comment:
It would be better to initialise this outside as doing it inside a function will result in creation of multiple instances

Reply:
Yes good catch, that was when I was still trying to make it work server side.